lib/common/patchset.c

```diff
@@ -1213,14 +1211,9 @@ apply_v2_patchset(xmlNode *xml, xmlNode *patchset)
                 rc = ENOMSG;
                 continue;
             }
-            while (pIter != NULL) {
-                const char *name = (const char *)pIter->name;
-
-                pIter = pIter->next;
-                xml_remove_prop(match, name);
-            }
+            pcmk__xe_remove_matching_attrs(match, NULL, NULL); // Remove all
 
-            for (pIter = pcmk__xe_first_attr(attrs); pIter != NULL;
+            for (xmlAttrPtr pIter = pcmk__xe_first_attr(attrs); pIter != NULL;
                  pIter = pIter->next) {
```

Comment:
would be another canditate to be functionalized

Reply:
Something like pcmk__xe_copy_attrs() to copy attrs from one element to another? It doesn't look like there are any other places that could use it, though there are several that are identical except using xmlSetProp() directly instead of crm_xml_add() (the main difference being ACL checks). I'm leaning to not messing with it.

Reply:
questionable if it is worth it but boolean check_acl would be a possibility ...